libminifi/include/core/state/nodes/AgentInformation.h

```diff
@@ -425,77 +427,121 @@ class AgentStatus : public StateMonitorNode {
-    } else {
-      uptime.value = "0";
+      serialized.push_back(serializeComponents());
     }
 
-    if (!repositories_.empty()) {
-      SerializedResponseNode repositories;
+    serialized.push_back(serializeResourceConsumption());
 
-      repositories.name = "repositories";
+    return serialized;
+  }
 
-      for (auto &repo : repositories_) {
-        SerializedResponseNode repoNode;
-        repoNode.collapsible = false;
-        repoNode.name = repo.first;
+ protected:
+  SerializedResponseNode serializeRepositories() {
+    SerializedResponseNode repositories;
 
-        SerializedResponseNode queuesize;
-        queuesize.name = "size";
-        queuesize.value = repo.second->getRepoSize();
+    repositories.name = "repositories";
 
-        SerializedResponseNode isRunning;
-        isRunning.name = "running";
-        isRunning.value = repo.second->isRunning();
+    for (auto &repo : repositories_) {
+      SerializedResponseNode repoNode;
+      repoNode.collapsible = false;
+      repoNode.name = repo.first;
 
-        SerializedResponseNode isFull;
-        isFull.name = "full";
-        isFull.value = repo.second->isFull();
+      SerializedResponseNode queuesize;
+      queuesize.name = "size";
+      queuesize.value = repo.second->getRepoSize();
 
-        repoNode.children.push_back(queuesize);
-        repoNode.children.push_back(isRunning);
-        repoNode.children.push_back(isFull);
-        repositories.children.push_back(repoNode);
-      }
-      serialized.push_back(repositories);
+      SerializedResponseNode isRunning;
+      isRunning.name = "running";
+      isRunning.value = repo.second->isRunning();
+
+      SerializedResponseNode isFull;
+      isFull.name = "full";
+      isFull.value = repo.second->isFull();
+
+      repoNode.children.push_back(queuesize);
+      repoNode.children.push_back(isRunning);
+      repoNode.children.push_back(isFull);
+      repositories.children.push_back(repoNode);
     }
+    return repositories;
+  }
 
-    serialized.push_back(uptime);
+  SerializedResponseNode serializeUptime() {
+    SerializedResponseNode uptime;
 
+    uptime.name = "uptime";
     if (nullptr != monitor_) {
+      uptime.value = monitor_->getUptime();
+    } else {
+      uptime.value = "0";
+    }
+
+    return uptime;
+  }
+
+  SerializedResponseNode serializeComponents() {
+    SerializedResponseNode components_node(false);
+    components_node.name = "components";
+    if (monitor_ != nullptr) {
       auto components = monitor_->getAllComponents();
-      SerializedResponseNode componentsNode(false);
-      componentsNode.name = "components";
 
       for (auto component : components) {
-        SerializedResponseNode componentNode(false);
-        componentNode.name = component->getComponentName();
+        SerializedResponseNode component_node(false);
+        component_node.name = component->getComponentName();
 
         SerializedResponseNode uuidNode;
```

Comment:
poor `uuidNode` got left behind with its old name 😏

Reply:
fixed in [d28d399](https://github.com/martinzink/nifi-minifi-cpp/commit/d28d3994fe633513158fb739edc49f69023b8a39)